packages/aws-library/tests/test_s3_client.py

```diff
@@ -0,0 +1,123 @@
+# pylint:disable=unused-variable
+# pylint:disable=unused-argument
+# pylint:disable=redefined-outer-name
+# pylint:disable=too-many-arguments
+# pylint:disable=no-name-in-module
+
+
+import csv
+import os
+from collections.abc import AsyncIterator
+from pathlib import Path
+
+import botocore.exceptions
+import pytest
+from aws_library.s3.client import SimcoreS3API
+from faker import Faker
+from models_library.api_schemas_storage import S3BucketName
+from moto.server import ThreadedMotoServer
+from pydantic import AnyUrl
+from pytest_simcore.helpers.utils_envs import EnvVarsDict
+from settings_library.s3 import S3Settings
+from types_aiobotocore_s3 import S3Client
+
+
+@pytest.fixture
+async def simcore_s3_api(
+    mocked_s3_server_settings: S3Settings,
+) -> AsyncIterator[SimcoreS3API]:
+    s3 = await SimcoreS3API.create(settings=mocked_s3_server_settings)
+    assert s3
+    assert s3.client
+    assert s3.exit_stack
+    assert s3.session
+    yield s3
+    await s3.close()
+
+
+async def test_aiobotocore_s3_client_when_s3_server_goes_up_and_down(
+    mocked_aws_server: ThreadedMotoServer,
+    mocked_s3_server_envs: EnvVarsDict,
+    s3_client: S3Client,
+):
+    # passes without exception
+    await s3_client.list_buckets()
+    mocked_aws_server.stop()
+    with pytest.raises(botocore.exceptions.EndpointConnectionError):
+        await s3_client.list_buckets()
```

Comment:
as I said, the whole s3_client was created not to raise botocore errors but our own via the s3_exception_handler. This should raise S3AccessError

Reply:
Will create a separate PR for this s3_client :+1: